src/integration/prepareRegistry.js

```diff
@@ -5,55 +5,69 @@ const cloner = require('../helpers/cloner'),
   concater = require('../helpers/concater'),
   creator = require('../helpers/creator'),
   misc = require('../helpers/misc'),
+  reader = require('../helpers/reader'),
   path = require('path');
 
 /**
  * This function gets the registry from a given location and combines into one json file.
  * You need to make sure to have a proper configuration settings in chewieConfig().js ('registry' attribute).
  * There are 2 different cases allowed, 'local' and 'remote'.
  * For 'remote' case, you need to pass a branch or tag name for the remote repository.
+ * @param {Object} [topics] - array with list of objects with topic name and type.
+ * Used in case you do not want to prepare registry with all topics but only with specific ones
  * @param {Object} [config] - basic integration configuration
  * @param {Function} [next] - callback for asynch operations
  */
 
 function prepareRegistry(topics, config, next) {
 
-  const registrySource = config.registry.path;
-  const registryOrigin = config.registry.location;
-  const branchTag = config.registry.branch;
+  const registrySource = config.registry.path,
+    registryOrigin = config.registry.location,
+    registryPath = config.registry.registryPath,
+    branchTag = config.registry.branch;
 
-  switch(registryOrigin){
+  //First step of preparing the registry is to actually check if it isn't already prepared
+  //If registry exists, we pass to callback
+  reader.readFile(registryPath, (err) => {
```

Comment:
We should use validator.fileCheck instead of reader.readFile. I believe that validator was created to do something like this :)